File: cookblog-frontend/src/app/api/resources/image.resource.ts
export class ImageResource {
  private readonly _id: number | null;
  private readonly _path: string;

  constructor(id: number, path: string) {
    this._id = id;
    this._path = path;
  }

  get id() {
    return this._id;
  }

  get path() {
    return this._path;
  }
}

File: cookblog-backend/src/main/java/com/example/cookblog/dto/resources/ImageResource.java
package com.example.cookblog.dto.resources;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Null;
import jakarta.validation.constraints.Size;
import lombok.Builder;
import lombok.extern.jackson.Jacksonized;

@Builder
@Jacksonized
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public record ImageResource(
        @Null Long id,
        @NotBlank @Size(max = 255) String path
) {

}